### DISCUSSION TITLE
Actions-Snake Animation

### DISCUSSION THREAD

**Discussion by ElvisAbner** [community/community#60370] [Actions] [answered]
I have tried everything and still in the push step I get this:

Run ad-m/github-push-action@master
Push to branch master
remote: Permission to ElvisAbner/ElvisAbner.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/ElvisAbner/ElvisAbner.git/': The requested URL returned error: 403
Error: Invalid exit code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ad-m/github-push-action/master/start.js:29:21)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 128
}
Error: Invalid exit code: 128
    at ChildProcess.<anonymous> (/home/runner/work/_actions/ad-m/github-push-action/master/start.js:29:21)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

This is my yml file: 

# GitHub Action for generating a contribution graph with a snake eating your contributions.

name: Generate Snake

# Controls when the action will run. This action runs every 6 hours.

on:
  schedule:
      # every 6 hours
    - cron: "0 */6 * * *"

# This command allows us to run the Action automatically from the Actions tab.
  workflow_dispatch:

# The sequence of runs in this workflow:
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:

    # Checks repo under $GITHUB_WORKSHOP, so your job can access it
      - uses: actions/checkout@v2

    # Generates the snake  
      - uses: Platane/snk@master
        id: snake-gif
        with:
          github_user_name: ElvisAbner
          # these next 2 lines generate the files on a branch called "output". This keeps the main branch from cluttering up.
          gif_out_path: dist/github-contribution-grid-snake.gif
          svg_out_path: dist/github-contribution-grid-snake.svg

     # show the status of the build. Makes it easier for debugging (if there's any issues).
      - run: git status

      # Push the changes
      - name: Push changes
        uses: ad-m/github-push-action@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          branch: master
          force: true

      - uses: crazy-max/ghaction-github-pages@v2.1.3
        with:
          # the output branch we mentioned above
          target_branch: output
          build_dir: dist
        env:
          ACCESS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
``

**Comment by Thesuspie** [comment, marked as answer]
In your personal repository, go to "**Settings**" on the left sidebar select "**Actions**" then inside the dropdown menu select "**General**" scroll down to "**Workflow permissions**" and toggle it to "**Read and write permissions**".

**Comment by ankit-0369** [comment]
I tried to add this snake animation in github read me file but in the output after running the workflow only 10% width of svg i.e. only two purple block has been generated how can I solve this

**Reply by iAdityaNarayanT** [reply]
I am facing the same issue

**Reply by Thesuspie** [reply]
I have had the same issue, i just updated my main.yml to use actions/checkout@v3, Platane/snk@v3 etc and now it works. Try modifying it according to my main.yml snake action. Or just copy it over and it should work. **NOTE** No need to change the value of **github_user_name** [property](https://github.com/Thesuspie/Thesuspie/blob/2d516dc6a3d70720d781138fabe30b84f57c5dc1/.github/workflows/main.yml#L33C11-L33C59) only modify **cron** [at this line](https://github.com/Thesuspie/Thesuspie/blob/2d516dc6a3d70720d781138fabe30b84f57c5dc1/.github/workflows/main.yml#L11) to change the interval if needed, or add a different colored output urls to the **outputs** at [this line](https://github.com/Thesuspie/Thesuspie/blob/2d516dc6a3d70720d781138fabe30b84f57c5dc1/.github/workflows/main.yml#L35)

**Reply by iAdityaNarayanT** [reply]
It worked! Thanks!

**Reply by rishika105** [reply]
It worked! thanks!
